quantum/wpm.c

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2020 Richard Sutherland (rich@brickbots.com)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "wpm.h"
+
+//WPM Stuff
+static uint8_t current_wpm = 0;
+static uint8_t latest_wpm = 0;
+static uint16_t wpm_timer = 0;
+
+//This smoothing is 40 keystrokes
+static const float wpm_smoothing = 0.0487;
+
+void set_current_wpm(uint8_t new_wpm) { current_wpm = new_wpm; }
+
+uint8_t get_current_wpm(void) { return current_wpm;}
+
+bool wpm_keycode(uint16_t keycode) {
+#    ifdef WPM_KEYCODE
+    return WPM_KEYCODE;
+#    else
+    return wpm_keycode_kb(keycode);
+#    endif
```

Comment:
I think there may be some issues with this logic.  It looks like if you define `WPM_KEYCODE`, it will trigger on every keystroke? 

Is that the intent?

Reply:
I suppose it depends on what you define it as... I was cribbing the pattern from the music mask example and was not 100% sure why the option was provided to use a define rather than overload the _kb or _user, so I included it as well.  

I'm happy to remove it to clarify and simplify the code.  I'll push a commit for that right now